swaylock/main.c

```diff
@@ -292,6 +294,10 @@ int main(int argc, char **argv) {
 	surfaces = create_list();
 	registry = registry_poll();
 
+	if (!registry) {
+		sway_abort("Unable to connect to sway");
```

Comment:
Yeah, that's fine for a different PR. Do any Wayland clients work at all in these conditions? Should sway set XDG_RUNTIME_DIR itself if not already set?